### ISSUE TITLE
Support event namespace

### ISSUE THREAD

**Issue by Johann-S** [sindresorhus/emittery#31]
It would be rad if it's possible for creators of libs to create a namespace for their events, it's well used in jQuery for example in Bootstrap.

Bootstrap Modal emit `show.bs.modal` event, it's allow user to be sure the `show` event come from the Bootstrap Modal and not an other lib or a general `show` event.

Currently, except jQuery I don't know an other libs which handle event namespace

See: 
 - https://api.jquery.com/event.namespace/
 - https://getbootstrap.com/docs/4.1/components/modal/#events

**Comment by sindresorhus** [comment]
You can already do this. Event namespaces in this sense is just a naming convention.

**Comment by Johann-S** [comment]
Yes but it's a bit different, for example, you have a DOM element which have a listener for `click.dom.element` but if user made a click on this element, we should call the listener because `click.dom.element` listen to the `click` event on this element

**Comment by kevva** [comment]
The difference between `emittery` and events emitted from the DOM is that they're global whilst events from `emittery` is tied to a specific instance. That's why I have a hard time seeing why this is needed.

**Comment by sindresorhus** [comment]
Closing this. As kevva noted, jQuery-style namespaces solve a problem specific to global event systems (like the DOM), where multiple libraries share the same element and need to selectively unbind handlers without affecting each other. Emittery is instance-scoped, so you get that isolation naturally by using separate instances for separate concerns.

If you want to group events by name, that is just a naming convention (`mouse.click`, `mouse.move`) and already works today. You can listen to multiple events at once with `on(["mouse.click", "mouse.move"], handler)` or use `onAny()` with filtering for broader patterns.
